shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureWrapper.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.renderer;
+
+import android.graphics.SurfaceTexture;
+import androidx.annotation.NonNull;
+
+/**
+ * A wrapper for a SurfaceTexture that tracks whether the texture has been released.
+ *
+ * <p>The engine calls {@code SurfaceTexture.release} on the platform thread, but {@code
+ * updateTexImage} is called on the raster thread. This wrapper will prevent {@code updateTexImage}
+ * calls on an abandoned texture.
+ */
+public class SurfaceTextureWrapper {
```

Comment:
This class needs the `@Keep` annotation if it's called from the native side.

Reply:
I also wonder if it makes sense to have this class as a inner class of `FlutterJNI`.

Reply:
I also wonder if it makes sense to have this class as a inner class of `FlutterJNI`.

Reply:
Added a `@Keep`.  `FlutterJNI` has become quite large and I thought it would be simpler to have a source file dedicated to just this purpose (similar to `FlutterOverlaySurface`)

Reply:
Got it. I agree. `FlutterJNI` is pretty big. One possible clean up is to have a `io.flutter.embedding.engine.jni` package, but I agree that is outside of scope for this change. The main value is that we could add linters to ensure `@Keep` is used for each class that belongs to that package, better testing patterns/policies around dependencies used by the JNI package, etc...